src/extension/fix_selinux_xattr/fix_selinux_xattr.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Author: Unrud
+ * Date:   18/12/2018
+ *
+ * Description: An extension that changes the return
+ * value of setxattr from EACCES to zero when trying
+ * to overwrite security.selinux with an identical value.
+ */
+
+#include <sys/xattr.h>
+#include "extension/extension.h"
+#include "tracee/mem.h"
+
+const unsigned int MAX_NAME_LENGTH = 17;
+
+/**
+ * Return zero instead of EACCES when trying
+ * to overwrite security.selinux with an identical value.
+ */
+static int handle_setxattr(Tracee *tracee)
+{
+    Sysnum sysnum = get_sysnum(tracee, ORIGINAL);
+    switch (sysnum) {
+    case PR_setxattr: 
+    case PR_lsetxattr:
+    case PR_fsetxattr: {
+        /* get the result of the syscall */
+        word_t res = peek_reg(tracee, CURRENT, SYSARG_RESULT);
+        if (res != -EACCES) {
+            return 0;
+        }
+
+        /* get the system call arguments */
+        word_t fd_or_path_start = peek_reg(tracee, MODIFIED, SYSARG_1);
+        word_t name_start = peek_reg(tracee, CURRENT, SYSARG_2);
+        word_t value_start = peek_reg(tracee, CURRENT, SYSARG_3);
+        word_t value_size = peek_reg(tracee, CURRENT, SYSARG_4);
+        word_t flags = peek_reg(tracee, CURRENT, SYSARG_5);
+
+        /* check if the attribute name is security.selinux */
+        char name[MAX_NAME_LENGTH];
+        int status = read_string(tracee, name, name_start, MAX_NAME_LENGTH);
+        if (status < 0) {
+            return status;
+        }
+        if (strncmp("security.selinux", name, MAX_NAME_LENGTH) != 0) {
+            return 0;
+        }
+
```

Comment:
It might be fine to just return **EPERM** here and drop the rest of the function.